File: rust/varar-runner/src/baseline_store.rs
//! The filesystem `BaselineStore`: the committed drift baseline lives at the
//! project root as `var.lock.json`. The core owns the format; this only reads
//! and writes the raw text.

use std::path::{Path, PathBuf};
use var_core::drift::BaselineStore;

pub struct FileBaselineStore {
    path: PathBuf,
}

impl FileBaselineStore {
    pub fn new(root: &Path) -> FileBaselineStore {
        FileBaselineStore {
            path: root.join("var.lock.json"),
        }
    }
}

impl BaselineStore for FileBaselineStore {
    fn read(&self) -> Option<String> {
        std::fs::read_to_string(&self.path).ok()
    }

    fn write(&mut self, contents: &str) {
        let _ = std::fs::write(&self.path, contents);
    }
}

File: go/runner/baseline_store.go
package runner

import (
	"os"
	"path/filepath"
)

// FileBaselineStore is the filesystem BaselineStore: the committed drift
// baseline lives at the project root as varar.lock.json. The core owns the
// format; this only reads and writes the raw text.
type FileBaselineStore struct {
	path string
}

// NewFileBaselineStore builds a store rooted at root/varar.lock.json.
func NewFileBaselineStore(root string) *FileBaselineStore {
	return &FileBaselineStore{path: filepath.Join(root, "varar.lock.json")}
}

// Read returns the lockfile contents, or ok=false when absent/unreadable.
func (s *FileBaselineStore) Read() (string, bool) {
	b, err := os.ReadFile(s.path)
	if err != nil {
		return "", false
	}
	return string(b), true
}

// Write persists the lockfile contents (best-effort).
func (s *FileBaselineStore) Write(contents string) {
	_ = os.WriteFile(s.path, []byte(contents), 0o644)
}
